--- a/src/panfrost/vulkan/panvk_vX_physical_device.c
+++ b/src/panfrost/vulkan/panvk_vX_physical_device.c
@@ -38,14 +38,14 @@
    const struct panvk_physical_device *device,
    struct vk_device_extension_table *ext)
 {
-   bool has_vk1_1 = PAN_ARCH >= 10;
-   bool has_vk1_2 = PAN_ARCH >= 10;
+   bool has_vk1_1 = PAN_ARCH >= 7;
+   bool has_vk1_2 = PAN_ARCH >= 7;
    bool has_gralloc = vk_android_get_ugralloc() != NULL;
 
    *ext = (struct vk_device_extension_table){
       .KHR_8bit_storage = true,
       .KHR_16bit_storage = true,
-      .KHR_shader_atomic_int64 = PAN_ARCH >= 9,
+      .KHR_shader_atomic_int64 = PAN_ARCH >= 7,
       .KHR_bind_memory2 = true,
       .KHR_buffer_device_address = true,
       .KHR_calibrated_timestamps = device->kmod.props.gpu_can_query_timestamp,
@@ -56,7 +56,7 @@
       .KHR_depth_clamp_zero_one = true,
       .KHR_depth_stencil_resolve = true,
       .KHR_device_group = true,
-      .KHR_draw_indirect_count = PAN_ARCH >= 10,
+      .KHR_draw_indirect_count = PAN_ARCH >= 7,
       .KHR_driver_properties = true,
       .KHR_dynamic_rendering = true,
       .KHR_dynamic_rendering_local_read = true,
@@ -108,7 +108,7 @@
       .KHR_shader_subgroup_rotate = true,
       .KHR_shader_subgroup_uniform_control_flow = has_vk1_1,
       .KHR_shader_terminate_invocation = true,
-      .KHR_spirv_1_4 = PAN_ARCH >= 10,
+      .KHR_spirv_1_4 = PAN_ARCH >= 7,
       .KHR_storage_buffer_storage_class = true,
 #ifdef PANVK_USE_WSI_PLATFORM
       .KHR_present_id2 = true,
@@ -136,7 +136,7 @@
 #ifdef VK_USE_PLATFORM_DISPLAY_KHR
       .EXT_display_control = true,
 #endif
-      .EXT_descriptor_indexing = PAN_ARCH >= 9,
+      .EXT_descriptor_indexing = PAN_ARCH >= 7,
       .EXT_extended_dynamic_state = true,
       .EXT_extended_dynamic_state2 = true,
       .EXT_external_memory_acquire_unmodified = true,
@@ -155,7 +155,7 @@
       .EXT_line_rasterization = true,
       .EXT_load_store_op_none = true,
       .EXT_non_seamless_cube_map = true,
-      .EXT_mutable_descriptor_type = PAN_ARCH >= 9,
+      .EXT_mutable_descriptor_type = PAN_ARCH >= 7,
       .EXT_physical_device_drm = true,
       .EXT_pipeline_creation_cache_control = true,
       .EXT_pipeline_creation_feedback = true,
@@ -164,8 +164,8 @@
       .EXT_primitive_topology_list_restart = true,
       .EXT_provoking_vertex = true,
       .EXT_queue_family_foreign = true,
-      .EXT_robustness2 = PAN_ARCH >= 10,
-      .EXT_sampler_filter_minmax = PAN_ARCH >= 10,
+      .EXT_robustness2 = PAN_ARCH >= 7,
+      .EXT_sampler_filter_minmax = PAN_ARCH >= 7,
       .EXT_scalar_block_layout = true,
       .EXT_separate_stencil_usage = true,
       .EXT_shader_module_identifier = true,
@@ -179,8 +179,8 @@
       .EXT_tooling_info = true,
       .EXT_vertex_attribute_divisor = true,
       .EXT_vertex_input_dynamic_state = true,
-      .EXT_ycbcr_2plane_444_formats = PAN_ARCH >= 10,
-      .EXT_ycbcr_image_arrays = PAN_ARCH >= 10,
+      .EXT_ycbcr_2plane_444_formats = PAN_ARCH >= 7,
+      .EXT_ycbcr_image_arrays = PAN_ARCH >= 7,
       .EXT_inline_uniform_block = true,
       .ANDROID_external_memory_android_hardware_buffer = has_gralloc,
       .ANDROID_native_buffer = has_gralloc,
@@ -254,7 +254,7 @@
       .sampleRateShading = true,
       .dualSrcBlend = true,
       .logicOp = true,
-      .multiDrawIndirect = PAN_ARCH >= 10,
+      .multiDrawIndirect = PAN_ARCH >= 7,
       .drawIndirectFirstInstance = true,
       .depthClamp = true,
       .depthBiasClamp = true,
@@ -275,7 +275,7 @@
       .vertexPipelineStoresAndAtomics =
          (PAN_ARCH >= 13 && instance->enable_vertex_pipeline_stores_atomics) ||
          instance->force_enable_shader_atomics,
-      .fragmentStoresAndAtomics = (PAN_ARCH >= 10) ||
+      .fragmentStoresAndAtomics = (PAN_ARCH >= 7) ||
           instance->force_enable_shader_atomics,
       .shaderTessellationAndGeometryPointSize = false,
       .shaderImageGatherExtended = true,
@@ -322,18 +322,18 @@
 
       /* Vulkan 1.2 */
       .samplerMirrorClampToEdge = true,
-      .drawIndirectCount = PAN_ARCH >= 10,
+      .drawIndirectCount = PAN_ARCH >= 7,
       .storageBuffer8BitAccess = true,
       .uniformAndStorageBuffer8BitAccess = true,
       .storagePushConstant8 = true,
-      .shaderBufferInt64Atomics = PAN_ARCH >= 9,
-      .shaderSharedInt64Atomics = PAN_ARCH >= 9,
-      .shaderFloat16 = PAN_ARCH >= 10,
+      .shaderBufferInt64Atomics = PAN_ARCH >= 7,
+      .shaderSharedInt64Atomics = PAN_ARCH >= 7,
+      .shaderFloat16 = PAN_ARCH >= 7,
       .shaderInt8 = true,
       /* In theory, update-after-bind is supported on bifrost, but the
        * descriptor limits would be too low for the descriptorIndexing feature.
        */
-      .descriptorIndexing = PAN_ARCH >= 9,
+      .descriptorIndexing = PAN_ARCH >= 7,
       .shaderInputAttachmentArrayDynamicIndexing = true,
       .shaderUniformTexelBufferArrayDynamicIndexing = true,
       .shaderStorageTexelBufferArrayDynamicIndexing = true,
@@ -344,17 +344,17 @@
       .shaderInputAttachmentArrayNonUniformIndexing = true,
       .shaderUniformTexelBufferArrayNonUniformIndexing = true,
       .shaderStorageTexelBufferArrayNonUniformIndexing = true,
-      .descriptorBindingUniformBufferUpdateAfterBind = PAN_ARCH >= 9,
-      .descriptorBindingSampledImageUpdateAfterBind = PAN_ARCH >= 9,
-      .descriptorBindingStorageImageUpdateAfterBind = PAN_ARCH >= 9,
-      .descriptorBindingStorageBufferUpdateAfterBind = PAN_ARCH >= 9,
-      .descriptorBindingUniformTexelBufferUpdateAfterBind = PAN_ARCH >= 9,
-      .descriptorBindingStorageTexelBufferUpdateAfterBind = PAN_ARCH >= 9,
-      .descriptorBindingUpdateUnusedWhilePending = PAN_ARCH >= 9,
-      .descriptorBindingPartiallyBound = PAN_ARCH >= 9,
+      .descriptorBindingUniformBufferUpdateAfterBind = PAN_ARCH >= 7,
+      .descriptorBindingSampledImageUpdateAfterBind = PAN_ARCH >= 7,
+      .descriptorBindingStorageImageUpdateAfterBind = PAN_ARCH >= 7,
+      .descriptorBindingStorageBufferUpdateAfterBind = PAN_ARCH >= 7,
+      .descriptorBindingUniformTexelBufferUpdateAfterBind = PAN_ARCH >= 7,
+      .descriptorBindingStorageTexelBufferUpdateAfterBind = PAN_ARCH >= 7,
+      .descriptorBindingUpdateUnusedWhilePending = PAN_ARCH >= 7,
+      .descriptorBindingPartiallyBound = PAN_ARCH >= 7,
       .descriptorBindingVariableDescriptorCount = true,
       .runtimeDescriptorArray = true,
-      .samplerFilterMinmax = PAN_ARCH >= 10,
+      .samplerFilterMinmax = PAN_ARCH >= 7,
       .scalarBlockLayout = true,
       .imagelessFramebuffer = true,
       .uniformBufferStandardLayout = true,
@@ -489,7 +489,7 @@
       /* VK_EXT_robustness2 */
       .robustBufferAccess2 = PAN_ARCH >= 11,
       .robustImageAccess2 = false,
-      .nullDescriptor = PAN_ARCH >= 10,
+      .nullDescriptor = PAN_ARCH >= 7,
 
       /* VK_KHR_shader_clock */
       .shaderSubgroupClock = device->kmod.props.gpu_can_query_timestamp,
@@ -517,10 +517,10 @@
       .texelBufferAlignment = true,
 
       /* VK_EXT_ycbcr_2plane_444_formats */
-      .ycbcr2plane444Formats = PAN_ARCH >= 10,
+      .ycbcr2plane444Formats = PAN_ARCH >= 7,
 
       /* VK_EXT_ycbcr_image_arrays */
-      .ycbcrImageArrays = PAN_ARCH >= 10,
+      .ycbcrImageArrays = PAN_ARCH >= 7,
 
       /* VK_EXT_non_seamless_cube_map */
       .nonSeamlessCubeMap = true,
@@ -531,7 +531,7 @@
       .unifiedImageLayoutsVideo = false,
 
       /* VK_EXT_mutable_descriptor_type */
-      .mutableDescriptorType = PAN_ARCH >= 9,
+      .mutableDescriptorType = PAN_ARCH >= 7,
 
 #ifdef PANVK_USE_WSI_PLATFORM
       /* VK_KHR_present_id2 */
@@ -545,7 +545,7 @@
       .deviceMemoryReport = true,
 
       /* VK_ARM_shader_core_builtins */
-      .shaderCoreBuiltins = PAN_ARCH >= 9,
+      .shaderCoreBuiltins = PAN_ARCH >= 7,
    };
 }
 
@@ -556,7 +556,7 @@
    if (version_override)
       return version_override;
 
-   if (PAN_ARCH >= 10)
+   if (PAN_ARCH >= 7)
       return VK_MAKE_API_VERSION(0, 1, 4, VK_HEADER_VERSION);
 
    return VK_MAKE_API_VERSION(0, 1, 0, VK_HEADER_VERSION);
@@ -565,7 +565,7 @@
 static VkConformanceVersion
 get_conformance_version()
 {
-   if (PAN_ARCH == 10)
+   if (PAN_ARCH == 7)
       return (VkConformanceVersion){1, 4, 1, 2};
 
    return (VkConformanceVersion){0, 0, 0, 0};
@@ -913,8 +913,8 @@
                                       VK_RESOLVE_MODE_MAX_BIT,
       .independentResolveNone = true,
       .independentResolve = true,
-      .filterMinmaxSingleComponentFormats = PAN_ARCH >= 10,
-      .filterMinmaxImageComponentMapping = PAN_ARCH >= 10,
+      .filterMinmaxSingleComponentFormats = PAN_ARCH >= 7,
+      .filterMinmaxImageComponentMapping = PAN_ARCH >= 7,
       .maxTimelineSemaphoreValueDifference = INT64_MAX,
       .framebufferIntegerColorSampleCounts = sample_counts,
 
@@ -1116,7 +1116,7 @@
       VK_IMAGE_LAYOUT_PREINITIALIZED,
 
       /* Only if vk1.1+ is supported */
-#if PAN_ARCH >= 10
+#if PAN_ARCH >= 7
       /*  Vulkan 1.1 */
       VK_IMAGE_LAYOUT_DEPTH_READ_ONLY_STENCIL_ATTACHMENT_OPTIMAL,
       VK_IMAGE_LAYOUT_DEPTH_ATTACHMENT_STENCIL_READ_ONLY_OPTIMAL,
